### DISCUSSION TITLE
how to use compose of transforms correctly?

### DISCUSSION THREAD

**Discussion by lg2578** [Project-MONAI/MONAI#4339] [Q&A]
I'm using compose of transforms,and there are some errors. I can't find any tips from the api docs. 
for examples:
```
trans=T.Compose([
    T.LoadImage(image_only=True),
    
    T.AddChannel(),
    T.Spacing(pixdim=(1.50,1.50)),
    T.Rotate90(),
    T.CropForeground(),
    T.Resize(spatial_size=(160,160),mode='bilinear'),
    T.EnsureType(),
   
])
```

- T.Rotate90()  lead to error:ValueError: Axes=(1, 2) out of range for array of ndim=2.

trans=T.Compose([
    T.LoadImage(image_only=True),
    
    T.AddChannel(),
    T.Spacing(pixdim=(1.50,1.50)),
    T.CropForeground(select_fn=lambda x: x>30),
    T.Resize(spatial_size=(160,160),mode='bilinear'),
    T.EnsureType(),
   
])

- T.Resize(spatial_size=(160,160),mode='bilinear'), lead to error:Input and output sizes should be greater than 0, but got input (H: 0, W: 1) output (H: 160, W: 160)
- if x>0 in T.CropForeground(select_fn=lambda x: x>30), T.Resize works

are there some principles,I need to know?

**Comment by Nic-Ma** [comment]
Hi @lg2578 ,

Thanks for your interest here.
May I know your MONAI version?
I think the default axes of `Rotate90` is (0, 1) is the latest version.
The `select_fn` of CropForeground is to define what's "foreground", seems your dataset uses `intensity > 30` as foreground?

Thanks.

**Reply by lg2578** [reply]
my version  is 0.8.1。
default settings for  CropForeground is x>0 ,and it works，I just want to try how about intensity > 30,and get an error. it seems when intensity > 30 the output size is H:1 W:0,it's not the truth.

**Reply by Nic-Ma** [reply]
I am afraid your test image may have no intensity > 30, I will test this corner case and get back to you soon.

Thanks.

**Reply by Nic-Ma** [reply]
Hi @lg2578 ,

I tried to execute some test programs locally, didn't see any similar issue.
Could you please help share the value range of your image? Especially the pixel map of intensity > 30?

Thanks.
